.config/agents/rules/mastra.md

```diff
@@ -1,11 +1,26 @@
 ---
-type: "agent_requested"
-description: "Use when working with Mastra (any package starting with `@mastra/*`)"
+name: mastra
+description: Mastra TypeScript framework for AI agents with memory, tools, workflows, and RAG. Use when working with @mastra/* packages or building AI agents.
+license: AGPL-3.0
+metadata:
+  triggers:
+    type: domain
+    enforcement: suggest
+    priority: medium
+    keywords:
+      - mastra
+      - "@mastra/"
+      - mastra.ai
+    intent-patterns:
+      - "\\b(use|build|create|implement)\\b.*?\\bmastra\\b"
+      - "\\b(agent|workflow|rag)\\b.*?\\bmastra\\b"
```

Comment:
@indietyp I'm curious about these properties: do they really affect the way the agent uses the skill? The "agent skills" standard seems to rather recommend sticking to the `description` as the means of hinting to the agent about when to load the skill; and they offer this [context injection pattern with their own python util](https://agentskills.io/integrate-skills#injecting-into-context) to provide more imperative

Reply:
The metadata field is not read by the agent that's correct. But I changed it so that the skill-rules.json file is now derived from the SKILL.md to colocate it. I am also happy to rewrite/use the Anthropic code/skill instead but didn't want to disturb the current hook based approach for anyone who is using it!